rocAL/rocAL/include/pipeline/tensor.h

```diff
@@ -145,17 +177,19 @@ class rocalTensorInfo {
     std::vector<size_t> _dims;  //!< denotes the dimensions of the tensor
     unsigned _batch_size;       //!< the batch size
     RocalMemType _mem_type;     //!< memory type, currently either OpenCL or Host
-    RocalROIType _roi_type;     //!< ROI type, currently either XYWH or LTRB
+    RocalROIType _roi_type = RocalROIType::XYWH;     //!< ROI type, currently either XYWH or LTRB
     RocalTensorDataType _data_type = RocalTensorDataType::FP32;  //!< tensor data type
     RocalTensorlayout _layout = RocalTensorlayout::NONE;     //!< layout of the tensor
     RocalColorFormat _color_format;  //!< color format of the image
-    std::shared_ptr<std::vector<RocalROI>> _roi;
-    size_t _data_type_size = tensor_data_size(_data_type);
-    size_t _data_size = 0;
+    void *_roi_buf;
+    uint64_t _data_type_size = tensor_data_size(_data_type);
+    uint64_t _data_size = 0;
     std::vector<size_t> _max_shape;  //!< stores the the width and height dimensions in the tensor
-    void reallocate_tensor_roi_buffers();
+    void allocate_tensor_roi_buffers();
+    void reset_tensor_roi_buffers();
     bool _is_image = false;
     bool _is_metadata = false;
+    size_t _channels = 3;   //!< stores the channel dimensions in the tensor
```

Comment:
Why is the value of channels hardcoded to 3 ?

Reply:
It is not hardcoded, but initialized to 3